# Now You Code 1: Data Analysis of Movie Goers

In this assignment you will perform a data analysis of people who go to the movies. 

A movie theatre chain asked movie goers to fill out a quick survey in exchange for a 1/2 price ticket. The survey asked for basic demographics: age, gender, occupation and zip code. This survey results are contained in the data file `'NYC1-moviegoers.csv'`

In this assignment you will write a series of Python pandas code (in several cells) to answer some basic questions about the responses in the dataset.


In [12]:
# this turns off warning messages
import warnings
warnings.filterwarnings('ignore')
from IPython.display import display

### Part 1: Load the dataset

write code to import pandas and load the dataset (in csv format) into the variable `moviegoers` and then print a random sample of 5 people from the data set.

In [16]:
import pandas as pd

moviegoers = pd.read_csv('NYC1-moviegoers.csv')
display(moviegoers.sample(5))

user_id  age gender     occupation zip_code
272      273   50      F          other    10016
310      311   32      M     technician    73071
777      778   34      M        student     1960
276      277   35      F  administrator    55406
675      676   30      M     programmer    32712

### Part 2: Gender distribution 

How many males and females filled out our survey? 

Write a single line of Python Pandas code to count the genders in the data set. (There should be M = 670, F = 273)

**HINT:** Select the `gender` column then use a built-in series method to count the values in the series.

In [24]:
moviegoers['gender'].value_counts()
moviegoers_sex_count = pd.DataFrame(moviegoers['gender'].value_counts())
display(moviegoers_sex_count)

gender
M     670
F     273

### Part 3: People without jobs

Who are the survey respondents without jobs? 

Write Python Pandas code to create a variable `no_occupation` which filters the `moviegoers` data set to only those survey respondents with an ocupation of `'none'`.  (There should be 9 people)


In [13]:
no_occupation = moviegoers[(moviegoers.occupation == "none")]
display(no_occupation)

user_id  age gender occupation zip_code age_group
56        57   16      M       none    84010     Youth
126      127   33      M       none    73439     Adult
129      130   20      M       none    60115     Adult
255      256   35      F       none    39042     Adult
288      289   11      M       none    94619     Youth
417      418   55      F       none    21206     Adult
627      628   13      M       none    94306     Youth
656      657   26      F       none    78704     Adult
755      756   30      F       none    90247     Adult

### Part 4: Gender distribution of people without jobs.

What is the gender distribution of the 9 respondents without jobs? 

Write Python Pandas code to display this. 

**HINT:** Use the variable `no_occupation` from the previous step.

In [26]:
no_occupation['gender'].value_counts()
no_occupation_sex_count = pd.DataFrame(no_occupation['gender'].value_counts())
display(no_occupation_sex_count)

gender
M       5
F       4

### Part 5: Young Artists

Write Python Pandas code to display the count of respondents with an occupation of artist who are 21 and under. (There should be 5)

**HINT:** You can either set each Pandas filter to a new `DataFrame` variable or try to chain the filters together. Also display them before you try and count them.

In [14]:
artists = moviegoers[(moviegoers.occupation == "artist") & (moviegoers.age <= 21)]
display(artists)

user_id  age gender occupation zip_code age_group
23        24   21      F     artist    94533     Adult
149      150   20      F     artist     2139     Adult
595      596   20      M     artist    77073     Adult
600      601   19      F     artist    99687     Adult
781      782   21      F     artist    33205     Adult

### Part 6: Distribution by age group

The movie theater which conducted this survey prices their tickets by age group:

 - Youth (age 18 and under) $7.50 
 
 - Adult (age 19 55) $12.50 
 
 - Senior (age 56 and up) $8.50 

Write python code to count the number of moviegoers in each of these age groups.

Your counts should be as follows:
```
Adult     837
Youth      54
Senior     52
```

**HINT:** You must perform feature engineering. Create a new column `'age_group'` and use the `'age'` column to assign one or more values to the age group. After you create the column and set the values get a count of values for the `'age_group'` column.



In [20]:
moviegoers['age_group'] = ''
moviegoers['age_group'][moviegoers['age'] <=18] = 'Youth'
moviegoers['age_group'][(moviegoers['age'] >=19) & (moviegoers['age'] <=55)] = 'Adult'
moviegoers['age_group'][moviegoers['age'] >=56] = 'Senior'
ages = moviegoers['age_group'].value_counts()
aged = pd.DataFrame(ages)
display(aged)

age_group
Adult         837
Youth          54
Senior         52

## Step 7: Questions

1. Pandas programs are different than typical Python programs. Explain the process by which you got the final solution?
    I got the final solution by adding a new column call age_group in the moviegoers panda. I then added three different age groups to this column called youth, adult and senoir that has conditions int hem that when compared with the age column information would sort a person into on of these three groups. I then made a new variable called age and did a value count of the new age_group column that gave me the number of people in each of the three different age ranges.
2. What was the most difficult aspect of this assignment? 
    I think the most difficult part of the assignment was the final solution. That is because addding the column and then created the three groups on that column was a bit different that what we have done before.

## Reminder of Evaluation Criteria

1. Was the problem attempted (analysis, code, and answered questions) ?
2. Was the problem analysis thought out? (does the program match the plan?)
3. Does the code execute without syntax error?
4. Does the code solve the intended problem?
5. Is the code well written? (easy to understand, modular, and self-documenting, handles errors)